functests/performance/hugepages.go

```diff
@@ -0,0 +1,120 @@
+package performance
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	centosImage = "centos:latest"
+	pathAvailableHugepages = "/sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages"
+	pathFreeHugepages = "/sys/devices/system/node/node0/hugepages/hugepages-1048576kB/free_hugepages"
+	pathUsageHugepages = "/rootfs/sys/fs/cgroup/hugetlb/hugetlb.1GB.usage_in_bytes"
+)
+
+var _ = Describe("[performance] Hugepages", func() {
+	var workerRTNode *corev1.Node
+
+	BeforeEach(func() {
+		workerRTNodes, err := nodes.GetByRole(testclient.Client, testutils.RoleWorkerRT)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(workerRTNodes).ToNot(BeEmpty())
+		workerRTNode = &workerRTNodes[0]
+	})
+
+	It("Verify available hugepages on the idle node", func() {
+		By("checking if the number of available and free hugepages is equal")
+		availableHugepages := checkHugepagesStatus(pathAvailableHugepages, workerRTNode)
+		freeHugepages := checkHugepagesStatus(pathFreeHugepages, workerRTNode)
+		Expect(availableHugepages).To(Equal(freeHugepages))
+
+		By("checking hugepages usage")
+		usageHugepages := checkHugepagesStatus(pathUsageHugepages, workerRTNode)
+		Expect(usageHugepages).To(Equal(0))
+	})
+
+	It("Pod should be able to run process with using hugepages", func() {
+		var testpod *corev1.Pod
+		podCommand := []string{"/bin/bash", "-c", "yum install -y libhugetlbfs-utils libhugetlbfs && " +
+			"LD_PRELOAD=libhugetlbfs.so HUGETLB_MORECORE=yes top -b > /dev/null"}
+
+		By("running the POD and waiting while it's installing testing tools")
+		testpod = getCentosPod(podCommand)
+		testpod, err := testclient.Client.Pods(testutils.NamespaceTesting).Create(testpod)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() string {
+			logs, err := pods.GetLogs(testclient.Client, testpod)
+			if err != nil {
+				return ""
+			}
+			return logs
+		}, 5*time.Minute, 10*time.Second).Should(ContainSubstring("Complete!"))
```

Comment:
can we pre-install these tools somehow? 5 mins seem long :)

Reply:
It's just timeout, in real this test takes around 1 minute. 
Unfortunately, I didn't find any containers with pre-installed libhugetlbfs tool..